c_glib/gandiva-glib/function-registry.cpp

```diff
@@ -0,0 +1,179 @@
+ */
+
+typedef struct GGandivaFunctionSignaturePrivate_ {
+  std::shared_ptr<gandiva::FunctionRegistry> *function_registry;
+  GList *native_function_list;
+} GGandivaFunctionRegistryPrivate;
+
+enum {
+  PROP_FUNCTION_REGISTRY = 1
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GGandivaFunctionRegistry,
+                           ggandiva_function_registry,
+                           G_TYPE_OBJECT)
+
+#define GGANDIVA_FUNCTION_REGISTRY_GET_PRIVATE(obj)      \
+    static_cast<GGandivaFunctionRegistryPrivate *>(      \
+        ggandiva_function_registry_get_instance_private( \
+          GGANDIVA_FUNCTION_REGISTRY(obj)))
+
+static void
+ggandiva_function_registry_dispose(GObject *object)
+{
+  G_OBJECT_CLASS(ggandiva_function_registry_parent_class)->dispose(object);
+}
+
+static void
+ggandiva_function_registry_finalize(GObject *object)
+{
+  auto priv = GGANDIVA_FUNCTION_REGISTRY_GET_PRIVATE(object);
+
+  if (priv->native_function_list) {
+    GList *p = priv->native_function_list;
+    while (p) {
+      g_object_unref(G_OBJECT(p->data));
+      p = g_list_next(p);
+    }
+    g_list_free(priv->native_function_list);
+  }
+
+  delete priv->function_registry;
+
+  G_OBJECT_CLASS(ggandiva_function_registry_parent_class)->finalize(object);
+}
+
+static void
+ggandiva_function_registry_set_property(GObject *object,
+                                        guint prop_id,
+                                        const GValue *value,
+                                        GParamSpec *pspec)
+{
+  auto priv = GGANDIVA_FUNCTION_REGISTRY_GET_PRIVATE(object);
+  switch (prop_id) {
+  case PROP_FUNCTION_REGISTRY:
+    {
+      auto ptr = g_value_get_pointer(value);
+      auto gandiva_function_registry
+        = static_cast<std::shared_ptr<gandiva::FunctionRegistry> *>(ptr);
+      priv->function_registry
+        = new std::shared_ptr<gandiva::FunctionRegistry>(*gandiva_function_registry);
+    }
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_function_registry_init(GGandivaFunctionRegistry *object)
+{
+}
+
+static void
+ggandiva_function_registry_class_init(GGandivaFunctionRegistryClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->dispose = ggandiva_function_registry_dispose;
+  gobject_class->finalize = ggandiva_function_registry_finalize;
+  gobject_class->set_property = ggandiva_function_registry_set_property;
+
+  GParamSpec *spec;
+  spec = g_param_spec_pointer("function_registry",
+                              "FunctionRegistry",
+                              "The raw gandiva::FunctionRegistry *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_FUNCTION_REGISTRY, spec);
+}
+
+/**
+ * ggandiva_function_registry_new:
+ *
+ * Returns: A newly created #GGandivaFunctionRegistry.
```

Comment:
Could you add `Since: 0.14.0`?

Reply:
Could you also add entry for `0.14.0` to `c_glib/doc/gandiva-glib/gandiva-glib-docs.xml`?

```diff
diff --git a/c_glib/doc/gandiva-glib/gandiva-glib-docs.xml b/c_glib/doc/gandiva-glib/gandiva-glib-docs.xml
index 2335e110..099270ab 100644
--- a/c_glib/doc/gandiva-glib/gandiva-glib-docs.xml
+++ b/c_glib/doc/gandiva-glib/gandiva-glib-docs.xml
@@ -68,6 +68,10 @@
     <title>Index of deprecated API</title>
     <xi:include href="xml/api-index-deprecated.xml"><xi:fallback /></xi:include>
   </index>
+  <index id="api-index-0-14-0" role="0.14.0">
+    <title>Index of new symbols in 0.14.0</title>
+    <xi:include href="xml/api-index-0.14.0.xml"><xi:fallback /></xi:include>
+  </index>
   <index id="api-index-0-12-0" role="0.12.0">
     <title>Index of new symbols in 0.12.0</title>
     <xi:include href="xml/api-index-0.12.0.xml"><xi:fallback /></xi:include>
```